actions/workflows/st2workroom_st2installer_st2enterprise_test.yaml

```diff
@@ -0,0 +1,134 @@
+---
+  vars:
+    root_dir: "/opt/puppet"
+    st2_username: "cibuild"
+    st2_password: "cibuild"
+  chain:
+    -
+      name: "create_vm_role"
+      ref: "st2cd.create_vm_role"
+      params:
+        hostname: "{{hostname}}"
+        instance_type: "{{instance_type}}"
+        environment: "{{environment}}"
+        key_name: "{{key_name}}"
+        keyfile: "{{keyfile}}"
+        distro: "{{distro}}"
+        role: "{{role}}"
+      publish:
+        distro: "{{distro}}"
+      on-success: "delete_existing_puppet_directory"
+    -
+      name: "delete_existing_puppet_directory"
+      ref: "linux.rm"
+      params:
+        hosts: "{{hostname}}"
+        target: "{{root_dir}}"
+        force: true
+        recursive: true
+        sudo: true
+      on-success: "write_answers_file"
+    -
+      name: "write_answers_file"
+      ref: "core.remote_sudo"
+      params:
+        hosts: "{{hostname}}"
+        # Note: We don't specify enterprise license key and other info.
+        # This information is provided via the installer.
+        cmd: |
+            cat <<EOT >> /tmp/answers.yaml
+            st2::version: {{version}}
+            st2::revision: {{build}}
+            st2::repo_base: {{dl_server}}
+            EOT
+      on-success: "bootstrap_workroom_with_answers_file"
+    -
+      name: "bootstrap_workroom_with_answers_file"
+      ref: "core.remote_sudo"
+      params:
+        env:
+          ENV: "{{branch}}"
+          SKIP_OK_CHECK: "true"
+          DISABLE_HUBOT: "true"
+          CI: "true"
+        hosts: "{{hostname}}"
+        cmd: "curl -sSL https://install.stackstorm.com/  | sudo sh -s -- -a /tmp/answers.yaml"
+        timeout: 3600
+      on-success: "verify_no_default_keys_are_installed_for_stanley_user"
+      on-failure: "rerun_bootstrap_workroom_with_answers_file"
+    -
+      name: "rerun_bootstrap_workroom_with_answers_file"
+      ref: "core.remote_sudo"
+      params:
+        env:
+          ENV: "{{branch}}"
+          SKIP_OK_CHECK: "true"
+          DISABLE_HUBOT: "true"
+          CI: "true"
+        hosts: "{{hostname}}"
+        cmd: "curl -sSL https://install.stackstorm.com/  | sudo sh -s -- -a /tmp/answers.yaml"
+        timeout: 3600
+      on-success: "verify_no_default_keys_are_installed_for_stanley_user"
+    -
+      name: "verify_no_default_keys_are_installed_for_stanley_user"
+      ref: "st2cd.st2workroom_verify_default_stanley_ssh_keys_are_not_installed"
+      params:
+        hosts: "{{hostname}}"
+        sudo: true
+      on-success: "run_installer"
+    -
+      name: "run_installer"
```

Comment:
As I mentioned on Slack - we want to test a clean state and not test a VM which is already provisioned.

And we want to test two scenarios from a clean state:

1. Non GUI installer where info is provided in answers.yaml (existing enterprise tests)
2. GUI based installer where part of the info is provided via the installer and second part of the provision is done by the installer (those new tests)

So in short - it would be very different because at that point the VM is already fully provisioned and I don't want to test an already fully provisioned VM - that's not a goal of those new tests.